### PR TITLE
Google services readme

### PR DESCRIPTION
**Pull request by 1fish2**
Also add the debug google-services.json to .gitignore.

---
This change is [Reviewable](https://reviewable.io/reviews/the-blue-alliance/the-blue-alliance-android/762)

### CODE REVIEW

**Comment by 1fish2** [comment]
Those merge commits are weird.

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,9 @@
 # Keys
 *.jks
 
+# Google Secrets
+android/src/debug/google-services.json
```
Comment (phil-lopreiato) [inline .gitignore]: I believe this is covered [in master](https://github.com/the-blue-alliance/the-blue-alliance-android/blob/master/.gitignore#L13)?

**Comment by phil-lopreiato** [comment]
Thanks for updating this - as you can tell, this is in a little bit of flux right now.

I really want to make this optional and for the app to be able to function with only the stubbed Google Services as found [here](https://github.com/the-blue-alliance/the-blue-alliance-android/blob/master/android/google-services.json), but the Firebase SDKs don't seem to have a good grasp on the 'optional' concept. For example, #761 instacrashes without a properly formed file.

File: .gitignore (hunk 1 above)
Comment (1fish2) [inline .gitignore]: Fixed. (I hate git.)

File: README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Want to add features, fix bugs, or just poke around the code? No problem!
 ### Project Management 
  - Keep up to date with the [mailing list](https://groups.google.com/forum/#!forum/thebluealliance-developers).
  - Read through the [planning documents](https://drive.google.com/#folders/0B5RO2Yzh2z01MDBOVXYwM1lXdFk) so you know what's going on.
- - Watch our [Trello board](https://trello.com/b/x42paPe3/tba-android) for updates on our long-term plans.
+ - Watch our [Trello board](https://trello.com/b/x42paPe3/tba-android) for updates on our long-term plans. [***TODO:*** Is this still relevant?]
```
Comment (phil-lopreiato) [inline README.md]: Nope

File: README.md [hunk 3]
```diff
@@ -51,6 +51,16 @@ Environment Setup
...
 5. If you have an Android device to test on, make sure to [enable USB Debugging](http://stackoverflow.com/questions/16707137/how-to-find-and-turn-on-usb-debugging-mode-on-nexus-4) in its Settings menu. Otherwise, [configure an Android Virtual Device (AVD)](https://developer.android.com/tools/devices/managing-avds.html) to debug with. To use a virtual device, you'll have to download an Android System image via the SDK manager. Android x86-based System images run much faster than ARM-based System images but they require [Virtual Machine Acceleration](http://developer.android.com/tools/devices/emulator.html#accel-vm). Note that the SDK manager will install the Intel HAXM _installer_ on your local disk; you still have to run that installer unless the Android Studio installer [does it for you](http://developer.android.com/tools/studio/index.html#install-updates).
+6. Set up a Firebase project to enable login authentication:
+    1. Open the [Firebase console](https://console.firebase.google.com/).
+    1. Create a Firebase project.
+        [***TODO:*** If you have a TBA dev web site AppEngine project, it may ask about using that project. Accept that offer?]
```
Comment (phil-lopreiato) [inline README.md]: Yes, my test rig has my app engine project linked. Although you may want to add it's not required.

File: README.md [hunk 4]
```diff
@@ -51,6 +51,16 @@ Environment Setup
...
+    1. Click "Add Firebase to your Android app".
+    1. Enter the package name `com.thebluealliance.androidclient.development` into the form.
+    1. Follow [these instructions](https://developers.google.com/android/guides/client-auth) to determine your debug SHA1. Enter that into the form.
+    1. Continue. It should download a `google-services.json` file.
+    1. Move the downloaded `google-services.json` file into `android/src/debug/`. ***Note:*** _Don't_ put it in the directory shown in the "Add Firebase to your Android app" instructions.
+    1. [***TODO:*** How to test this setup?]
```
Comment (phil-lopreiato) [inline README.md]: Put in that you should be able to build and run the app, for now. If you have `FirebaseCrash` in use, it'll throw an exception in app init if the file is invalid.

**Comment by phil-lopreiato** [comment]
Thanks @1fish2!

**Pull request merged**